CefSharp.Core/Internals/CefFrameWrapper.cpp

```diff
@@ -0,0 +1,235 @@
+// Copyright � 2010-2015 The CefSharp Project. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+#include "Stdafx.h"
+
+#include "Internals/CefSharpBrowserWrapper.h"
+#include "Internals/CefFrameWrapper.h"
+
+///
+// True if this object is currently attached to a valid frame.
+///
+/*--cef()--*/
+bool CefFrameWrapper::IsValid()
+{
+    return _frame->IsValid();
+}
+
+///
+// Execute undo in this frame.
+///
+/*--cef()--*/
+void CefFrameWrapper::Undo()
+{
+    _frame->Undo();
+}
+
+///
+// Execute redo in this frame.
+///
+/*--cef()--*/
+void CefFrameWrapper::Redo()
+{
+    _frame->Redo();
+}
+
+///
+// Execute cut in this frame.
+///
+/*--cef()--*/
+void CefFrameWrapper::Cut()
+{
+    _frame->Cut();
+}
+
+///
+// Execute copy in this frame.
+///
+/*--cef()--*/
+void CefFrameWrapper::Copy()
+{
+    _frame->Copy();
+}
+
+///
+// Execute paste in this frame.
+///
+/*--cef()--*/
+void CefFrameWrapper::Paste()
+{
+    _frame->Paste();
+}
+
+///
+// Execute delete in this frame.
+///
+/*--cef(capi_name=del)--*/
+void CefFrameWrapper::Delete()
+{
+    _frame->Delete();
+}
+
+///
+// Execute select all in this frame.
+///
+/*--cef()--*/
+void CefFrameWrapper::SelectAll()
+{
+    _frame->SelectAll();
+}
+
+///
+// Save this frame's HTML source to a temporary file and open it in the
+// default text viewing application. This method can only be called from the
+// browser process.
+///
+/*--cef()--*/
+void CefFrameWrapper::ViewSource()
+{
+    _frame->ViewSource();
+}
+
+///
+// Retrieve this frame's HTML source as a string sent to the specified
+// visitor.
+///
+/*--cef()--*/
+void CefFrameWrapper::GetSource(IStringVisitor^ visitor)
```

Comment:
Probably nicer to wrap these in a `Task` like the `ChromiumWebBrowser` instances do, same for `GetText()`